### PR TITLE
Remove unused file

### PR DESCRIPTION
**Pull request by sirreal**
#6317 copied the file to state/purchases/reducer, but I haven't tracked down where the change to use the other reducer was made.

However, if the app can build with no issues without this file it should be safe to remove.

## Testing
Smoke test app, especially  [`/me/purchases`](https://calypso.live/me/purchases?branch=remove/lib-purchases-reducer)

### CODE REVIEW

File: client/lib/purchases/reducer.js [hunk 1]
```diff
@@ -1,165 +0,0 @@
...
-
-		// the purchase doesn't match the predicate or is missing from the array of new purchases
-		return false;
-	} );
-}
-
-function updatePurchases( existingPurchases, action ) {
-	let purchases, predicate;
-
-	if ( ActionTypes.PURCHASES_SITE_FETCH_COMPLETED === action.type ) {
```
Comment (tyxla) [inline client/lib/purchases/reducer.js]: Seems like many of those legacy actions (or probably all of them that start with `PURCHASE`) are unused but still defined in https://github.com/Automattic/wp-calypso/blob/master/client/lib/upgrades/constants.js. Perhaps we could also clean those up?

Comment (sirreal) [reply client/lib/purchases/reducer.js]: I took a look and it seems like all of the following (removed in this PR) are used elsewhere:

```
PRIVACY_PROTECTION_CANCEL_COMPLETED
PRIVACY_PROTECTION_CANCEL_FAILED
PURCHASE_REMOVE_COMPLETED
PURCHASE_REMOVE_FAILED
PURCHASES_REMOVE
PURCHASES_SITE_FETCH
PURCHASES_SITE_FETCH_COMPLETED
PURCHASES_SITE_FETCH_FAILED
PURCHASES_USER_FETCH
PURCHASES_USER_FETCH_COMPLETED
PURCHASES_USER_FETCH_FAILED
```

Comment (tyxla) [reply client/lib/purchases/reducer.js]: Most of these are actually not used - the confusion probably comes that for some of them we have the same names, defined as Redux action types in `client/state/action-types.js`.

Comment (sirreal) [reply client/lib/purchases/reducer.js]: 🤦‍♂️ You're probably right… this is nasty duplication. I'll try to track some down.

**Pull request merged**